### PR TITLE
use bulkRemove for OpenSearch index cleanup

### PR DESCRIPTION
**Pull request by yeganathan18**
# Changes proposed ✍️

### What
copilot:summary
​
copilot:poem

### How
copilot:walkthrough

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [x] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: services/libs/opensearch/src/service/member.sync.service.ts [hunk 1]
```diff
@@ -135,9 +135,7 @@ export class MemberSyncService {
 
       if (toRemove.length > 0) {
         this.log.warn({ tenantId, toRemove }, 'Removing members from index!')
-        for (const id of toRemove) {
-          await this.openSearchService.removeFromIndex(id, OpenSearchIndex.MEMBERS)
-        }
+        await this.openSearchService.bulkRemoveFromIndex(toRemove, OpenSearchIndex.MEMBERS)
```
Comment (themarolt) [inline services/libs/opensearch/src/service/member.sync.service.ts:138]: maybe go even further and gather ids in toRemove until you reach the maximum allowed for bulk remove?

**Review by themarolt — CHANGES_REQUESTED** [review]

File: services/libs/opensearch/src/service/member.sync.service.ts (hunk 1 above)
Comment (yeganathan18) [reply services/libs/opensearch/src/service/member.sync.service.ts:138]: Good point! There's no limit for bulk removal in OpenSearch; we can do as much as we want, but that also affects the time taken to do the operation on the passed IDs.

I thought this should be safe and quick 🤔

Comment (themarolt) [reply services/libs/opensearch/src/service/member.sync.service.ts:138]: What I mean was:
currently we do cleanup by `tenantId` like this:
```
while (nextPage) {
  memberIds = fetchFromOpensearch
  memberIdsNotInDb = filterThroughDb(memberIds)
  delete memberIdsNotInDb from OpenSearch
}  
```

what if we do smt like this:
```
memberIdsNotInDb = []
while (nextPage) {
  memberIds = fetchFromOpensearch
  memberIdsNotInDb.push(filterThroughDb(memberIds))
  if (memberIdsNotInDb.length > SOME_THRESHOLD) {
      delete memberIdsNotInDb from OpenSearch
      memberIdsNotInDb = []
   }
} 

if (memberIdsNotInDb.length > 0) {
 delete memberIdsNoInDb from OpenSearch
} 
```

this is to eliminate too many OpenSearch delete/bulkDelete calls - maybe test it out a bit to see if it pays out on the long run for bigger tenants...

Comment (yeganathan18) [reply services/libs/opensearch/src/service/member.sync.service.ts:138]: Ah gotcha, maybe we can have `SOME_THRESHOLD` as 300.

**Comment by yeganathan18** [comment]
@themarolt Since we already cleaned up the organizations in OpenSearch, we couldn't test this properly as it's already pretty fast with fewer items to remove 🙂

But it's much better than before!

**Review by themarolt — APPROVED** [review]

**Pull request merged**